packages/editor/src/store/selectors.js

```diff
@@ -782,6 +782,26 @@ export function getBlockRootClientId( state, clientId ) {
 	return null;
 }
 
+/**
+ * Given a block client ID, returns the root of the hierarchy from which the block is
+ * nested, return the block itself for root level blocks.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId Block from which to find root client ID.
+ *
+ * @return {string} Root client ID
+ */
+export function getBlockHierarchyRootClientId( state, clientId ) {
```

Comment:
This in combination with `getBlockRootClientId` isn't very optimized, particularly when a consuming UI is always present on the screen. Mentioning because I've seen it come up already as one of the higher time-wasters in performance audit results. I might think we should try to do one of:

- Avoid calling the selector if possible (e.g. only when dropdown open)
- Memoize one or both selectors
- Reshape state to optimize upward-path traversal
   - This one is actually fairly straight-forward with adding a `rootClientId` property to the block state. I've only been reluctant because it's technically redundant, but I think performance would win out here.